cdc/sink/producer/kafka/kafka.go

```diff
@@ -116,9 +119,10 @@ func (c *Config) Initialize(sinkURI *url.URL, replicaConfig *config.ReplicaConfi
 	c.ClientID = sinkURI.Query().Get("kafka-client-id")
 
 	s = sinkURI.Query().Get("protocol")
-	if s != "" {
-		replicaConfig.Sink.Protocol = s
+	if s == "" {
+		return cerror.ErrKafkaInvalidConfig.GenWithStack("encoding protocol is not set")
 	}
```

Comment:
Is this a breaking change, if TiCDC upgrades and an old changefeed whose protocol is not set in sink-uri, the changefeed can't be resumed without changing it sink-uri.

Reply:
got it, we have to handle it in another way.